tensor/dense.go

```diff
@@ -267,3 +342,178 @@ func (t *Dense) shallowClone() *Dense {
 	}
 	return retVal
 }
+
+/* ------ Mask operations */
+
+//ResetMask fills the mask with either false, or the provided boolean value
+func (t *Dense) ResetMask(val ...bool) error {
+	if !t.IsMasked() {
+		t.makeMask()
+	}
+	var fillValue = false
+	if len(val) > 0 {
+		fillValue = val[0]
+	}
+	memsetBools(t.mask, fillValue)
+	return nil
+}
+
+// HardenMask forces the mask to hard. If mask is hard, then true mask values can not be unset
+func (t *Dense) HardenMask() bool {
+	t.maskIsSoft = false
+	return t.maskIsSoft
+}
+
+// SoftenMask forces the mask to soft
+func (t *Dense) SoftenMask() bool {
+	t.maskIsSoft = true
+	return t.maskIsSoft
+}
+
+// MaskFromSlice makes mask from supplied slice
+func (t *Dense) MaskFromSlice(x interface{}) {
```

Comment:
This is more a curiosity than anything - is there any reason why there are so many construction options for `MaskFromSlice` ? Wouldn't a `[]bool` suffice?